Cogburn/sharing sessions

coreyogburn opened the pull request:
When viewing a session in the OnionAI UI, there's a new checkbox that toggles the `shared` tag on the session. This will serve a larger purpose in later tickets such as attaching sessions to cases.

coreyogburn committed:
Toggle Session Shared Tag

This takes the first steps towards conversations being accessible by other users. The original owner of a session can mark a session as shared by tagging it. In later commits, we'll use this tag when determining if a user can access a session that isn't theirs.

The UI is ugly but functional.

Tests are needed.

coreyogburn committed:
Stable Working State

Ripped out a lot of the GetSessions logic that checks perms ahead of time. Now it executes the search and filters the result set. No longer need to indicate before hand if the search is looking for authored sessions or not.

Fix existing tests so they pass. Will look into adding more after some cleanup.

coreyogburn committed:
Unbroke Investigations

If GetSessionDetails returns a 404, then investigations won't execute.

coreyogburn committed:
UI Tweaks and Styles

coreyogburn committed:
Prevent Non-Owner Changes, Tests

Keep track of who owns which sessions.

Disable the Shared button when the session is not owned. Require the session to be owned when toggling the Shared checkbox. Both of these prevent the button from getting out of sync.

Added a constant for a session tag.

Added tests.

jertel commented on html/pages/assistant.html:
```diff
@@ -102,7 +102,7 @@ <h4 data-aid="assistant_credits_remaining">{{ i18n.creditsRemaining }}: <span :c
 					<span class="flex-grow-1">{{ i18n.chatTitle }}</span>
 					<div v-if="currentChatId">
-						<v-checkbox :model-value="((chatHistoryById[currentChatId] || {}).tags || []).includes('shared')" @change="toggleSharedSession()" density="compact" class="mx-4" hide-details :label="i18n.shareSessionLabel"></v-checkbox>
+						<v-checkbox :model-value="((chatHistoryById[currentChatId] || {}).tags || []).includes('shared')" :disabled="$root.user.id !== (chatHistoryById[currentChatId] || {}).userId" @change="toggleSharedSession()" density="compact" class="mx-4" hide-details :label="i18n.shareSessionLabel"></v-checkbox>
```
Missing data-aid

coreyogburn replied:
Fixed, changed to link/broken-link icon and moved left.

jertel replied:
I still don't see the `data-aid` though.

jertel replied:
Nevermind, I was looking at a stale version.

jertel commented on server/assistanthandler.go:
```diff
@@ -438,7 +439,7 @@ func (h *AssistantHandler) GetSessions(w http.ResponseWriter, r *http.Request) {
+// @Summary      Get Session Details
 // @Description  Retrieve the complete chat history for a specific session.
 // @Tags         Assistant
 // @Security     bearer[assistant/read_authored]
```
How will you communicate the read_shared permission requirements?

coreyogburn replied:
Added a new `@Security` line

jertel commented on server/assistanthandler.go:
```diff
@@ -468,15 +469,109 @@ func (h *AssistantHandler) GetSessionHistory(w http.ResponseWriter, r *http.Requ
+	})
+}
+
+func (h *AssistantHandler) UpdateSession(w http.ResponseWriter, r *http.Request) {
```
API Docs?

coreyogburn replied:
Added

coreyogburn committed:
Respond to feedback

Refactor Shared checkbox to Link/Unlink icon and move it to the left.

Refactor Compress Context button to be an icon to the right of the Context Length.

Add link icon to sessions in Chat History that are shared.

Update/Add documentation for endpoints.

coreyogburn committed:
Remove bulky debug log
